java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture2/tasks/_2/Number2.java

```diff
@@ -0,0 +1,229 @@
+package com.zoolatech.lecture2.tasks._2;
+
+
+/*Define a class that represents a robot, which moves in a room of a width W and a height of H.
+The class should provide methods to move a robot forward by either 1 or N tiles and turn it either left or right
+by 90 degrees from the current direction.
+The class should also provide methods to get the robot's current location and direction.
+The north-west corner of a room is represented by coordinates
+(0,0); north-east - (W-1, 0); south-east - (W-1, H-1); south-west - (0, H-1).
+By default, a robot should be placed in the north-west corner facing south direction.
+However, it should be also possible to give any coordinates and direction as the initial
+robot location. Robot must not be able to get outside of a room - in such case the robot should
+move as much as possible to the room border, print a warning message and stop.
+All next moves in the same direction should be prevented.
+*/
+
+import java.util.Arrays;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class Number2 {
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        RobotGame robotGame = new RobotGame();
+        boolean exit = false;
+        System.out.println("HELLO AND WELCOME!. Please input a W(width)");
+        robotGame.setW(scanner.nextInt());
+        System.out.println("and H(height) of a room");
+        robotGame.setH(scanner.nextInt());
+        System.out.println("Your room size is:");
+        robotGame.getRoomSize();
+        System.out.println("Press 0 to continue");
+        scanner.nextInt();
+        System.out.println();
+        System.out.println("Please input a coordinates for robot.\nThere must be smaller than your room :)");
+        System.out.println("Input X and then Y");
+        robotGame.setCoordinates(scanner.nextInt(), scanner.nextInt());
+        System.out.println("Your coordinates are:");
+        robotGame.getCoordinates();
+        System.out.println();
+        System.out.println("Press 0 to continue");
+        System.out.println();
+        System.out.println("Set direction where your robot is going");
+        System.out.println("0. SOUTH,\n1. NORTH,\n2. WEST,\n3. EAST");
+        Direction direction = Direction.values()[Integer.parseInt(scanner.next())];
+
+        switch (direction.toString()) {
```

Comment:
You don't need to have a switch statement here - you can use `direction` variable instead to display its value and to the robot.